tests/unit-tests/conversions.cpp

```diff
@@ -0,0 +1,62 @@
+#include "doctest/doctest.h"
+
+#include <ipr/impl>
+
+TEST_CASE("C++ Standard Conversions") {
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module module{lexicon};
+  impl::Interface_unit unit{lexicon, module};
+
+  // static_cast<long long>(4);
+  //    Integral Promotion            (Promote)
+  lexicon.make_promote(
+    *lexicon.make_literal(lexicon.int_type(), "4"),
+    &lexicon.long_long_type()
+  );
+
+  // (float)2.2;
+  //    Floating-Point Conversion     (Demote)
+  lexicon.make_demote(
+    *lexicon.make_literal(lexicon.double_type(), "2.2"),
+    &lexicon.float_type()
+  );
+
+  // (volatile int)7;
+  //    CV-Qualifcation               (Qualification)
+  lexicon.make_qualification(
+    lexicon.get_qualified(Type_qualifier::Volatile, lexicon.int_type()),
+    *lexicon.make_literal(lexicon.double_type(), "2.2"),
+    &lexicon.int_type() // Type of the expr can differ from target type (see 7.2.2/2)
+  );
+
+  // int* const ptr = 0;
+  //    Pointer Conversion            (Coersion)
+  const auto& ptr_type = lexicon.get_qualified(
+    Type_qualifier::Const, lexicon.get_pointer(lexicon.int_type()));
+  auto& ptr = *unit.global_region()->declare_var(*lexicon.make_identifier("ptr"), ptr_type);
+  ptr.init = lexicon.make_coerce(
+    ptr_type,
+    *lexicon.make_literal(lexicon.int_type(), "0"),
+    &ptr_type
+  );
+
+  // if (ptr) double(6);
+  //    Boolean Conversion            (Coersion)
+  //    Lvalue-to-Rvalue              (Read)
+  //    Integral-Floating Conversion  (Coersion)
+  const auto& condition = *lexicon.make_coerce(
+    lexicon.bool_type(),
+    *lexicon.make_read(
+      *lexicon.make_id_expr(ptr),
+      &lexicon.get_pointer(lexicon.int_type()) // != ptr.type() (see 7.2.2/2)
```

Comment:
I'm not sure the standard citation adds anything here. The citation describes automatic cv adjustment, but I think integer literals are always prvalues of type `T`, not `cv T`. (Based on https://eel.is/c++draft/lex.literal#lex.icon-3). I'd recommend either removing the citation or making the inner pointer type something that is cv qualified.

Reply:
In this the type of `ptr` is a const pointer not a pointer to a const int. I choose this example to illustrate that the type of the expression may not be the same as the type of the expr child. @GabrielDosReis This is something I would also like clarification because I am not 100% sure I am doing the right thing.

Reply:
It is a good example.